app/post.go

```diff
@@ -1668,3 +1668,18 @@ func (a *App) GetPostIfAuthorized(postID string, session *model.Session) (*model
 
 	return post, nil
 }
+
+func (a *App) GetPostsByIds(postIDs []string) ([]*model.Post, *model.AppError) {
+	posts, err := a.Srv().Store.Post().GetPostsByIds(postIDs)
+	if err != nil {
+		var nfErr *store.ErrNotFound
+		switch {
+		case errors.As(err, &nfErr):
```

Comment:
The store API never returns a `store.ErrNotFound` error. But it's a good idea if you want to change the store method to return this error. It's only called in searchlayer/post_layer.go, so it should be fine.

Reply:
Add a not found error to the store if no posts are found